apps/src/feedback.js

```diff
@@ -1133,13 +1134,8 @@ FeedbackUtils.prototype.getShowCodeComponent_ = function(
   challenge = false
 ) {
   const numLinesWritten = this.getNumBlocksUsed();
-  const shouldShowTotalLines =
-    options.response &&
-    options.response.total_lines &&
-    options.response.total_lines !== numLinesWritten;
-  const totalNumLinesWritten = shouldShowTotalLines
-    ? options.response.total_lines
-    : 0;
+  const totalLines = clientState.lines();
+  const totalNumLinesWritten = totalLines !== numLinesWritten ? totalLines : 0;
```

Comment:
I see this logic is similar to what was here before, but I'm not sure I understand it.  Could you briefly describe what the intent here is?

Reply:
Oops. Turns out I added some buggy code. 
What this (now) does: In the success dialog, display the total lines of code ever written by the user. If the user is logged in, use the total lines stored on the server. Otherwise, read it from the client. Was that what you were wondering, or were you asking about the ternary operator?

Reply:
I was wondering about both.  The calculation for totalLines makes sense now, but I'm still not sure about the calculation for totalNumLinesWritten?  (Maybe I'm misunderstanding what totalNumLinesWritten means?)

Reply:
Ahh ok. I get what you're asking now. This is definitely a confusing section of code.
totalLines: count of all lines a user has ever written
numLinesWritten: count of lines written in this current level
totalNumLinesWritten: count of all lines a user has ever written. Unless if this is the first time a user has written code (only reason totalLines would equal numLinesWritten). Then we set it to zero. Which gets passed into CodeWritten which decides whether to display a message with the count of all lines a user has ever written based on if totalNumLinesWritten is zero.

Ok that was way too hard to explain. I'm going to refactor :)